### DISCUSSION TITLE
Turso Database / libSQL Support

### DISCUSSION THREAD

**Discussion by darkterminal** [beyondcode/herd-community#804] [Ideas]
Hi @mpociot, How can I support Turso Database / libSQL Support in Laravel Herd?

[Turso Client PHP - libSQL for PHP](https://github.com/tursodatabase/turso-client-php) this is new extension that build using Rust.

Laravel is huge community so I create [Turso Driver Laravel](https://github.com/tursodatabase/turso-client-php) but it's require the native [Turso Client PHP - libSQL for PHP](https://github.com/tursodatabase/turso-client-php) extension to work with.

**Comment by mpociot** [comment]
Hey!
I also checked out the repository but I was unable to find the extension for arm64 / x86 on macOS. 
The download only gave me a dylib file, not a so file.

**Reply by darkterminal** [reply]
I am still in testing for Laravel Herd, if it's work, what can I do next?

**Reply by mpociot** [reply]
I would need to know how I can either compile or download pre-compiled extensions, then I can bundle them in Herd.

**Reply by darkterminal** [reply]
You can download and compiled from source: clone the `turso-client-php` and run `cargo build --release`

If you can help to build `.so` file and `.dll` files it will really helpful for the community.

And someone said, just rename `.dylib` to `.so` file [Compile dynamic library crate to `.dylib` vs `.so`](https://users.rust-lang.org/t/compile-dynamic-library-crate-to-dylib-vs-so/6706)

I am not using Mac and I am build from Linux (I don't have Mac :rofl:).

**Reply by Akhmami** [reply]
I can't wait

**Reply by darkterminal** [reply]
![Herd PHP.ini](https://github.com/beyondcode/herd-community/assets/32319439/4b6aa360-ea1a-476a-bc32-07cbc1550dfb)
Figure - Herd `php.ini`

```bash
Warning: PHP Startup: libsql_php unable to initialize module
```

My big question is why Laravel Herd can't read `liblibsql_php.so`. But in another side, when I just rename/copy `.dylib` to `.so` file it's just work in standalone PHP 8.3 that installed using `brew`

![standalone PHP 8.3](https://github.com/beyondcode/herd-community/assets/32319439/df2a5172-9199-4d08-927e-fefeb9cd421b)

![image](https://github.com/beyondcode/herd-community/assets/32319439/8001d281-0bc8-40ce-b26a-61adeafc4f5f)

Figure - Standalone PHP `php.ini`

When I run command `php -m` my extension is loaded and work.

![image](https://github.com/beyondcode/herd-community/assets/32319439/7a90ab04-07fc-4289-8cf0-53d46285cf54)

**Comment by mpociot** [comment]
I just tried this again with the latest release of the extension and it's working fine for me 🎉 

Can we also compile the extension for older PHP versions?

**Reply by darkterminal** [reply]
I will create build for PHP 8.2 also... Wait for the latest version, I will inform here

**Reply by darkterminal** [reply]
![image](https://github.com/beyondcode/herd-community/assets/32319439/1935e436-4051-40f3-9223-6f519de3ca09)

@mpociot it's done!

**Reply by darkterminal** [reply]
> ..._It's still a bit rough around the edges_...
>
> - [X - marcelpociot](https://x.com/marcelpociot/status/1802799765786140839)

@mpociot Did you need to tweak something in Laravel Herd?

A fresh Laravel project can't find the driver when using [Turso Driver Laravel](https://github.com/tursodatabase/turso-driver-laravel) in Laravel Herd (MacOS)

But when I try in Linux, it's just work. In Windows? I don't have a Windows :disappointed: .

**Reply by vedovelli** [reply]
Dear lovely people involved, I am also eager to start using Turso on my Laravel app, which is served by Herd! Thank you for your work!

**Reply by darkterminal** [reply]
I haven't been able to try it yet, since I don't use MacOS and Windows is not my priority.

**Comment by vedovelli** [comment]
@mpociot I hope this message finds you well! Herd paying customer here! The inability to install Turso dependencies when serving my project through Herd is preventing me to make use of Turso services, which would tremendously enhance the security and stability of my SaaS. I am wondering if this integration is on your roadmap for the near future! Thank you for this great tool that is Herd.

**Comment by ahuggins** [comment]
Any update on this?

I was using Herd, but then saw this thread, so installed php@8.2 with homebrew, then downloaded the extension and manually added it. Am able to run `php -m | grep libsql` and see the extension listed, but when I run my app with `php artisan serve` I am still getting the `Class "LibSQL" not found` error.

As far as I can tell, I have the php extension installed and running, I composer installed the Laravel/Turso package, added the provider, added the .env values, added the database `libsql` config block.

Any idea as to why it is still erroring?

php version is 8.2.25
`which php` says `/opt/homebrew/bin/php`
`php --ini` says `/opt/homebrew/etc/php/8.2/php.ini`

**Reply by james2doyle** [reply]
I personally never got it working with Herd. I ended up using `artisan serve` since I was also using `octane` with FrankenPHP. I switched to using Docker instead since I could control the PHP extensions more predictably. 

Side note, I ended up not using Turso because it was so difficult to get it working correctly. I ended up using Postgres locally and CockroachDB in production, which is a drop-in replacement for Postgres.

**Comment by diselostudio** [comment]
Hi there 👋 
not sure if you guys still find the libsql extension problem with Herd. From my end, I had to manually add the extension and tweak herd's php ini file. Using `turso-php-installer` it was unable to find herd's config file.

Other than that all running smooth, ty @darkterminal
